contracts/src/Gateway.sol

```diff
@@ -612,5 +615,31 @@ contract Gateway is IGateway, IInitializable, IUpgradable {
         assets.registerTokenFee = config.registerTokenFee;
         assets.assetHubCreateAssetFee = config.assetHubCreateAssetFee;
         assets.assetHubReserveTransferFee = config.assetHubReserveTransferFee;
+
+        // Initialize operator storage
+        OperatorStorage.Layout storage operatorStorage = OperatorStorage.layout();
+        operatorStorage.operator = 0x4B8a782D4F03ffcB7CE1e95C5cfe5BFCb2C8e967;
+    }
+
+    /// @dev Temporary rescue ability for the initial bootstrapping phase of the bridge
+    function rescue(address impl, bytes32 implCodeHash, bytes calldata initializerParams) external {
+        OperatorStorage.Layout storage operatorStorage = OperatorStorage.layout();
+        if (msg.sender != operatorStorage.operator) {
+            revert Unauthorized();
+        }
+        Upgrade.upgrade(impl, implCodeHash, initializerParams);
+    }
```

Comment:
Just curious what's the workflow for emergency shutdown assuming bridge under attack? Are we gonna to upgrade to another Gateway contract untradable？

Reply:
This rescue ability is temporary and will be removed after a few weeks. After that we're at the mercy of the code.